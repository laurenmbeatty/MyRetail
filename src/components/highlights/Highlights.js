import React from "react";
import * as Styled from "./styles";
class Highlights extends React.Component {
  render() {
    const { highlights } = this.props;
    return (
      <div>
        <Styled.HighlightsTitle>product highlights</Styled.HighlightsTitle>
        {highlights && (
          <Styled.Highlights>
            {highlights.map((highlight, index) => (
              <li key={index} dangerouslySetInnerHTML={{ __html: highlight }} />
            ))}
          </Styled.Highlights>
        )}
      </div>
    );
  }
}
export default Highlights;
