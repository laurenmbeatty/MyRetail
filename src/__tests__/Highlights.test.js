import React from "react";
import { create } from "react-test-renderer";
import { render, cleanup } from "react-testing-library";
import "jest-styled-components";
import Highlights from "../components/highlights/Highlights";

afterEach(cleanup);

test("snapshot", () => {
  const c = create(<Highlights />);
  expect(c.toJSON()).toMatchSnapshot();
});

test("it renders with some props", () => {
  const props = {
    features: [
      "<strong>Wattage Output:</strong> 1100 Watts",
      "<strong>Number of Speeds:</strong> 3 ",
      "<strong>Capacity (volume):</strong> 72.0 Oz.",
      "<strong>Appliance Capabilities:</strong> Blends",
      "<strong>Includes:</strong> Travel Lid",
      "<strong>Material:</strong> Plastic",
      "<strong>Finish:</strong> Painted",
      "<strong>Metal Finish:</strong> Chrome",
      "<strong>Safety and Security Features:</strong> Non-Slip Base",
      "<strong>Care and Cleaning:</strong> Easy-To-Clean, Dishwasher Safe Parts"
    ]
  };

  const { getByText } = render(<Highlights highlights={props.features} />);

  const highlight = getByText(/Number of Speeds:/);

  expect(highlight).toBeDefined();
});
