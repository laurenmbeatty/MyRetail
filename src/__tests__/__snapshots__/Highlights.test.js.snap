// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <h2
    className="sc-bdVaJa irPHUE"
  >
    product highlights
  </h2>
</div>
`;
